delphi-checks/src/test/java/au/com/integradev/delphi/checks/UnusedTypeCheckTest.java

```diff
@@ -127,4 +127,50 @@ void testHelperShouldNotAddIssue() {
                 .appendDecl("  end;"))
         .verifyNoIssues();
   }
+
+  @Test
+  void testUnusedApiTypeWithExcludeApiShouldNotAddIssue() {
```

Comment:
It might be good to add a test that a public nested type in the interface section is excluded, like

```delphi
// ...
interface

type
  TFoo = class(TObject)
  public type
    TBar = class(TObject)
    end;
  end;
```

I see there is already a test that non-public nested types are included.